lib/update/updater.js

```diff
@@ -35,24 +35,16 @@ Updater.factory = function factory (config) {
 Updater.prototype._fetchDeviceId = function _fetchDeviceId () {
   const netInterfaces = os.networkInterfaces()
 
-  let macAddrs = []
-  for (let [key, val] of Object.entries(netInterfaces)) {
-    val.forEach(item => {
-      if (item.mac !== '00:00:00:00:00:00') {
-          macAddrs.push({ name: [key], mac: item.mac })
-      }
-    })
-  }
-
-  macAddrs.sort((lhs, rhs) => {
-      if (lhs.mac < rhs.mac) return -1
-      else if (lhs.mac > rhs.mac) return 1
-      else return 0
-  })
-
-  if (macAddrs.length > 0) {
-    const name = macAddrs[0].name
-    const mac = macAddrs[0].mac.replace(/:/g, '-')
+  const mainInterface =
+    _(netInterfaces)
+    .pickBy((item) => item[0].mac !== '00:00:00:00:00:00')
+    .toPairs()
+    .sortBy((item) => item[1].mac)
+    .first()
```

Comment:
After ``toPairs``, we get an array of pairs. The second half of each pair is an array, possibly with only one value. So I don't think the sort will work here.

Here's what I came up with, although it loses the interface name, which we don't really need:

```
_(os.networkInterfaces())
  .values()
  .flatten()
  .reject(_.isMatch({mac: '00:00:00:00:00:00'}))
  .sortBy('mac')
  .map('mac')
  .first()
```